students/EricAdams/session07/html_render.py

```diff
@@ -3,7 +3,9 @@
 class Element():
 
     tag = 'html'
-    indent = '  '
+    indent = ''
+    # keep track of the depth of recursion
+    depth = 0
```

Comment:
ideally you don't need to do this at all, though it's a fine approach.

But you definitely don't want depth as a class attribute -- class attributes are for when all the intances should share the same value -- certainly not the case here!